From 76bc236e727bfb4c5249a2251457d586ba0bde5d Mon Sep 17 00:00:00 2001
From: Alf Henrik Sauge <alf.henrik.sauge@gmail.com>
Date: Thu, 14 May 2026 18:57:45 +0200
Subject: [PATCH] Fix crash in 'Filter by Path' due to out-of-bound access

---
 src/index/Index.cpp | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/index/Index.cpp b/src/index/Index.cpp
index 3179951..929099e 100644
--- a/src/index/Index.cpp
+++ b/src/index/Index.cpp
@@ -260,9 +260,13 @@ QList<git::Commit> Index::commits(const QList<Posting> &postings) const {
   // Look up commits.
   QSet<git::Commit> commits;
   foreach (const Posting &posting, postings) {
-    // FIXME: Remove deleted commits on write.
-    if (git::Commit commit = mRepo.lookupCommit(mIds.at(posting.id)))
-      commits.insert(commit);
+    // If we fail this check, then the index is mismatching the repo
+    // The FIXME below might be the cause of that
+    if (posting.id < mIds.size()) {
+      // FIXME: Remove deleted commits on write.
+      if (git::Commit commit = mRepo.lookupCommit(mIds.at(posting.id)))
+        commits.insert(commit);
+    }
   }
 
   return commits.values();
-- 
2.54.0

